lib/util/print.js

```diff
@@ -91,4 +96,25 @@ function printTreeTraversal(node) {
   }
 }
 
+// Need to add parens around any [+, -] operation
+// next to [/, *, ^] operation or parenthesis node
+// Examples:
+// { "/" [{ "+" ["x", "2"] }, "2"] } -> (x + 2) / 2.
+// { "+" [{ "+" ["x", "x"] }, { "+" ["2", "2"] }] } -> (x + x) + (2 + 2)
+// Check #120, #126 issues for more details.
+function needToAddParens(node, parentNode) {
+  if (!parentNode ||
+      !Node.Type.isOperator(parentNode) ||
+      '*/^'.indexOf(node.op) !== -1) {
```

Comment:
I find this a bit confusing, since we're wanting the parent to have + or -

I'd prefer `'+-'.indexOf(node.op) === -1`

ooo or alternatively, you could just do

```
if (!parentNode ||
    !Node.Type.isOperator(parentNode, '+') ||
    !Node.Type.isOperator(parentNode, '-') || {
```

not sure which one is cleaner - what do you like better?